runtime/service/src/main/java/org/apache/polaris/service/credentials/DefaultPolarisCredentialManager.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.service.credentials;
+
+import jakarta.annotation.Nonnull;
+import jakarta.enterprise.inject.Any;
+import jakarta.enterprise.inject.Instance;
+import org.apache.polaris.core.connection.AuthenticationType;
+import org.apache.polaris.core.connection.ConnectionConfigInfoDpo;
+import org.apache.polaris.core.credentials.PolarisCredentialManager;
+import org.apache.polaris.core.credentials.connection.ConnectionCredentialVendor;
+import org.apache.polaris.core.credentials.connection.ConnectionCredentials;
+import org.apache.polaris.service.credentials.connection.SupportsAuthType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default implementation of {@link PolarisCredentialManager} responsible for retrieving credentials
+ * used by Polaris to access external systems such as remote catalogs or cloud storage.
+ *
+ * <p>This implementation delegates to {@link ConnectionCredentialVendor} implementations selected
+ * via CDI based on the authentication type. Each vendor handles the credential transformation logic
+ * for a specific authentication mechanism (e.g., SigV4, OAuth).
+ *
+ * <p>Flow:
+ *
+ * <ol>
+ *   <li>Selects the appropriate {@link ConnectionCredentialVendor} based on the authentication type
+ *   <li>Delegates to the vendor to generate the final connection credentials (the vendor will
+ *       resolve the service identity internally)
+ * </ol>
+ */
+public class DefaultPolarisCredentialManager implements PolarisCredentialManager {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(DefaultPolarisCredentialManager.class);
+
+  private final Instance<ConnectionCredentialVendor> credentialVendors;
+
+  public DefaultPolarisCredentialManager(
+      @Any Instance<ConnectionCredentialVendor> credentialVendors) {
+    this.credentialVendors = credentialVendors;
+  }
+
+  @Override
+  public @Nonnull ConnectionCredentials getConnectionCredentials(
+      @Nonnull ConnectionConfigInfoDpo connectionConfig) {
```

Comment:
In the future, we may embed creds cache in the credential manager, if we can't find a valid entry in the cache, we will select the correct vendor and delegate the credential retrieval to the credential vendor (for both storage and connection).

In this way, we can deprecate the old flow (using metastore to get the subscoped creds).

Since we need to generate cache key and may need realm id and the catalog id, wandering if we should pass in the whole catalog entity here. But we can refactor this later once we add cache support for credential manager.

cc: @dimas-b @dennishuo @HonahX